components/bl-leaflet-map/src/helpers/map.js

```diff
@@ -103,18 +105,34 @@ export function createCircles(circles, map, eventHandlers) {
   }
 }
 
-export function createMarkers(markers, icon, map, eventHandlers) {
-  if (markers) {
-    const { onMarkerClick } = eventHandlers;
+export function useMarkers(markers, icon, map, eventHandlers) {
+  const [markersArray, setMarkersArray] = useState([]);
 
-    markers.forEach(item => {
-      Leaflet.marker([item.point.lat, item.point.lng], { icon })
-        .on('click', () => {
-          onMarkerClick({ coordinates: [item.point.lat, item.point.lng], description: item.description });
-        })
-        .addTo(map)
-        .bindPopup(item.description);
-    });
+  useEffect(() => {
+    clearOldMarkers(markersArray, setMarkersArray);
+
+    if (markers) {
+      const { onMarkerClick } = eventHandlers;
+
+      markers.forEach(({ point: { lng, lat }, description }) => {
+        const marker = Leaflet.marker([lat, lng], { icon })
+          .on('click', () => {
+            onMarkerClick({ coordinates: [lat, lng], description });
+          })
+          .addTo(map)
+          .bindPopup(description);
+
+        setMarkersArray(prev => [...prev, marker]);
```

Comment:
Could we use refs for avoid rerender from useState?

Also here you call setter as many times as you have markers, how about to do it once?